proxies/live/apiproxy/resources/jsc/validate-client-assertion-claims.js

```diff
@@ -0,0 +1,26 @@
+
+function extractJsonVariable(contextVariableName) {
+  return JSON.parse(
+    context.getVariable(
+      "jwt.DecodeJWT.FromClientAssertionFormParam."
+      + contextVariableName
+    )
+  )
+}
+
+
+jwtHeaders = extractJsonVariable('header-json')
+jwtPayload = extractJsonVariable('payload-json')
+
+// Set context variables based on the condition in the second argument
+context.setVariable('InvalidJwt.MissingKidHeader', !jwtHeaders.kid);
+context.setVariable('InvalidJwt.MissingOrInvalidTypHeader', typeof jwtHeaders.typ != "string" || jwtHeaders.typ.toLowerCase() != "jwt");
+context.setVariable('InvalidJwt.MissingExpClaim', !jwtPayload.exp);
+context.setVariable('InvalidJwt.InvalidExpiryTime', typeof jwtPayload.exp != "number");
```

Comment:
what if it is a number but not a valid date? e.g 1234?

Reply:
Every integer is a valid UTC timestamp as it's just the number of seconds since epoch. I thought about adding validation that the date must not be in the past, but this is already implemented:

```
        <FaultRule name="expired_jwt">
            <Condition>(JWT.failed Equals true) And (fault.name Equals "TokenExpired")</Condition>
            <Step>
                <Name>AssignMessage.ExpiredJWTError</Name>
            </Step>
        </FaultRule>
```